src/components/ChangeOracles.tsx

```diff
@@ -0,0 +1,286 @@
+      return 'Invalid address';
+    }
+  }, [addresses, currentInput]);
+
+  const removeAddress = useCallback(
+    address => {
+      setAddresses(addresses.filter(a => a !== address));
+    },
+    [addresses],
+  );
+
+  const addAddress = useCallback(() => {
+    setAddresses([...addresses, currentInput]);
+    setCurrentInput('');
+    addressInput.current.focus();
+  }, [addresses, currentInput]);
+
+  const canMakeProposal = !!charterOfferId && addresses.length > 0;
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    console.debug({ event, addresses, minutesUntilClose });
+    const offerFn = (() => {
+      switch (mode) {
+        case ChangeOraclesMode.Add:
+          return walletUtils.makeVoteOnAddOracles;
+        case ChangeOraclesMode.Remove:
+          return walletUtils.makeVoteOnRemoveOracles;
+      }
+    })();
+
+    const offer = offerFn(
+      charterOfferId,
+      priceFeed,
+      addresses,
+      minutesUntilClose,
+    );
+    void walletUtils.sendOffer(offer);
+  }
+
+  const addressesList = (
+    <ul>
+      <AnimatePresence>
+        {addresses.map(address => (
+          <motion.li
+            className="overflow-hidden"
+            key={address}
+            initial={{ height: 0, opacity: 0, marginBottom: 0 }}
+            animate={{ height: 'auto', opacity: 1, marginBottom: 8 }}
+            exit={{ height: 0, opacity: 0, marginBottom: 0 }}
+            transition={{ type: 'tween', duration: 0.25 }}
+          >
+            <ListItem
+              address={address}
+              onRemove={() => removeAddress(address)}
+            />
+          </motion.li>
+        ))}
+      </AnimatePresence>
+    </ul>
+  );
+
+  const submitButtonLabel = useMemo(() => {
+    switch (mode) {
+      case ChangeOraclesMode.Add:
+        return 'Propose Add Oracles';
+      case ChangeOraclesMode.Remove:
+        return 'Propose Remove Oracles';
+    }
+  }, [mode]);
+
+  const inputPlaceholder = useMemo(() => {
+    switch (mode) {
+      case ChangeOraclesMode.Add:
+        return 'Oracle address to add, i.e. something that looks like "agoric1XXXX..."';
+      case ChangeOraclesMode.Remove:
+        return 'Oracle address to remove, i.e. something that looks like "agoric1XXXX..."';
+    }
+  }, [mode]);
+
+  const content = priceFeed && (
+    <motion.div
+      className="overflow-hidden px-1"
+      initial={{ height: 0, opacity: 0 }}
+      animate={{ height: 'auto', opacity: 1 }}
+      transition={{ type: 'tween' }}
+    >
+      <h2 className="my-2 block text-lg leading-5 font-medium text-gray-700">
+        Operator Addresses{' '}
+        {addresses.length === 0 ? '' : `(${addresses.length})`}
+      </h2>
+      <div className="mt-4">{addressesList}</div>
+      <div className="p-2 rounded flex flex-row gap-2 border border-gray-300 focus-within:border-purple-300 focus-within:ring-purple-300 focus-within:ring-1">
+        <input
+          type="text"
+          ref={addressInput}
```

Comment:
random thought: you know those annoying login forms that don't let you paste a password? this one could do the opposite: not let you type and you have to paste :)